go.mod

```diff
@@ -1,25 +1,70 @@
 module github.com/doitintl/kube-no-trouble
 
-go 1.16
+go 1.19
 
 require (
-	github.com/Azure/go-autorest/autorest/adal v0.9.5 // indirect
-	github.com/DATA-DOG/go-sqlmock v1.4.1 // indirect
 	github.com/ghodss/yaml v1.0.0
-	github.com/google/gofuzz v1.1.0 // indirect
 	github.com/hashicorp/go-version v1.3.0
-	github.com/jmoiron/sqlx v1.2.0 // indirect
-	github.com/lib/pq v1.3.0 // indirect
 	github.com/open-policy-agent/opa v0.34.2
-	github.com/paultyng/changelog-gen v1.1.0 // indirect
 	github.com/rs/zerolog v1.21.1-0.20210413053206-582f0cf0e39b
-	github.com/rubenv/sql-migrate v0.0.0-20200402132117-435005d389bc // indirect
 	github.com/spf13/pflag v1.0.5
 	helm.sh/helm v2.17.0+incompatible
 	helm.sh/helm/v3 v3.1.3
 	k8s.io/apimachinery v0.17.13
 	k8s.io/client-go v0.17.13
+)
+
+require (
```

Comment:
A couple of questions about these changes:

- Where do they come from? New go version?
- Is it a standard practice to have two `require` blocks? _(I see it's a split between explicit and indirect)_
- Why are there so many new `indirect` dependencies?

Reply:
New changes to go mod. Since go1.17

https://go.dev/doc/go1.17#go-command

Reply:
> For the go command to correctly resolve transitive imports using the pruned module graph, the go.mod file for each module needs to include more detail about the transitive dependencies relevant to that module. If a module specifies go 1.17 or higher in its go.mod file, its go.mod file now contains an explicit require directive for every module that provides a transitively-imported package. (In previous versions, the go.mod file typically only included explicit requirements for directly-imported packages.)

Reply:
You can see example here: https://github.com/helm/helm/blob/main/go.mod

Reply:
I feel these changes are good because we are explicit about which packages we are using.

Reply:
Yep, makes sense 👍